### PR TITLE
(#71) Implemented Images.iterate() : Iterable<Image>

### PR DESCRIPTION
**Pull request by llorllale**
This PR:
* solves #71 by implementing `Images.iterate()`
* Uses the [Images API Docs](https://docs.docker.com/engine/api/v1.35/#operation/ImageList) as reference
* Leaves puzzle for the rest of the `Docker` interface
* Leaves puzzle for the rest of the `Images` interface
* Fixes puzzle format in Response.java.

### CODE REVIEW

**Comment by 0crat** [comment]
Job [#82](https://github.com/amihaiemil/docker-java-api/issues/82) is now in scope, role is `REV`

File: src/test/java/com/amihaiemil/docker/RtImagesTestCase.java [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+ */
+public final class RtImagesTestCase {
+    /**
+     * Must return the same number of images as there are elements in the
+     * json array returned by the service.
+     * @throws Exception If an error occurs.
+     */
+    @Test
+    public void iterateReturnsImages() throws Exception {
+        final AtomicInteger count = new AtomicInteger();
```
Comment (amihaiemil) [inline src/test/java/com/amihaiemil/docker/RtImagesTestCase.java]: @llorllale maybe a dumb question, but why is this an AtomicInteger?

File: src/main/java/com/amihaiemil/docker/RtImage.java [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+import java.net.URI;
+import org.apache.http.client.HttpClient;
+
+/**
+ * Runtime {@link Image}.
+ * @author George Aristy (george.aristy@gmail.com)
+ * @version $Id$
+ * @since 0.0.1
+ */
+final class RtImage implements Image {
```
Comment (amihaiemil) [inline src/main/java/com/amihaiemil/docker/RtImage.java]: @llorllale Can you also leave a puzzle here, to continue implementing operations on a single ``Image``?

File: src/main/java/com/amihaiemil/docker/RtImages.java [hunk 3]
```diff
@@ -0,0 +1,90 @@
...
+     * @param client The http client.
+     * @param uri The URI for this Images API.
+     */
+    RtImages(final HttpClient client, final URI uri) {
+        this.client = client;
+        this.baseUri = uri;
+    }
+
+    @Override
+    public Iterable<Image> iterate() throws IOException {
```
Comment (amihaiemil) [inline src/main/java/com/amihaiemil/docker/RtImages.java]: @llorllale This is nice, for now. But in the future, I would like ``Images`` itself to be an ``Iterable<Image>``, so you can simply iterate over the Images like this:

```
for(Image img : docker.images()) {
 //...
}
```

So basically, this iterating logic will be inside a custom Iterator which would fetch them at before starting to iterate.

Comment (llorllale) [reply src/main/java/com/amihaiemil/docker/RtImages.java]: @amihaiemil I like it. Will leave a puzzle.

Comment (llorllale) [reply src/main/java/com/amihaiemil/docker/RtImages.java]: @amihaiemil btw, it will mean we will have to swallow an `IOException`. Are you ok with that?

Comment (amihaiemil) [reply src/main/java/com/amihaiemil/docker/RtImages.java]: @llorllale Yes, I guess the tradeoff is worth it. And of course, we'll explain it in the javadoc.

Comment (llorllale) [reply src/main/java/com/amihaiemil/docker/RtImages.java]: @amihaiemil right

**Comment by llorllale** [comment]
@amihaiemil left puzzles as per your comments, please check

**Comment by amihaiemil** [comment]
@rultor merge

**Comment by 0crat** [comment]
The job [#82](https://github.com/amihaiemil/docker-java-api/issues/82) is now out of scope

**Pull request merged**